backend/services/operators.py

```diff
@@ -0,0 +1,191 @@
+import random
+from abc import ABC, abstractmethod
+from backend.models.individual import Individual
+from backend.models.chromosome import Chromosome
+
+# --- SELEKCJA ---
+
+class SelectionOperator(ABC):
+    @abstractmethod
+    def select(self, population, **kwargs):
+        """
+        Wybiera osobniki z populacji.
+        """
+        pass
+
+class TournamentSelection(SelectionOperator):
+    def __init__(self, tournament_size=3):
+        self.tournament_size = tournament_size
+
+    def select(self, population, **kwargs):
+        selected = []
+        pop = population.individuals
+        for _ in range(len(pop)):
+            tournament = random.sample(pop, self.tournament_size)
+            best = min(tournament, key=lambda ind: ind.fitness)  # Zakładamy minimalizację
+            selected.append(best)
```

Comment:
Czy to powinno działać tak, ze jak mamy N osobników, to selekcja wybierze N wyselekcjonowanych? 

Nie powinno być raczej tak, że dzielisz ich na grupy, a potem ze zwyciezców tworzysz kolejne grupy, aż będzie tylko M najlepszych?